### ISSUE TITLE
Wrong link_rewrite format but still can be POST via webserver

### ISSUE THREAD

**Issue by WilfredChong** [PrestaShop/PrestaShop#13151]
I encountered an error the Friendly URL, as if link_rewrite input is "abc.com" , Prestashop allows to create the product instead of giving an error as "invalid field value" for friendly URL. 

Below is the request's body for the same for which I use the post method to create the product: 
Request:

`<?xml version='1.0' encoding='UTF-8'?>
<prestashop>
	<product>
		<id_supplier>3</id_supplier>
		<id_manufacturer>3</id_manufacturer>
		<id_tax_rules_group>1</id_tax_rules_group>
		<width>10</width>
		<height>10</height>
		<depth>10</depth>
		<weight>10</weight>
		<id_category_default/>
		<id_shop_default>1</id_shop_default>
		<on_sale>1</on_sale>
		<ean13>1234567</ean13>
		<upc/>
		<price>889.38</price>
		<wholesale_price/>
		<ecotax/>
		<unit_price_ratio/>
		<unity/>
		<reference>Leather Bag Collection</reference>
		<active>1</active>
		<available_for_order>1</available_for_order>
		<condition>new</condition>
		<visibility>both</visibility>
		<show_price>1</show_price>
		<online_only>0</online_only>
		<name>
			<language id='1'>Leather Bag Collection</language>
		</name>
		<description>
			<language id='1'>Leather Bag Collection</language>
		</description>
		<description_short>
			<language id='1'>Short Desc</language>
		</description_short>
		<link_rewrite>
			<language id='1'>xyz.com</language>
		</link_rewrite>
		<meta_title>
			<language id='1'>Meta</language>
		</meta_title>
		<meta_description>
			<language id='1'>Meta</language>
		</meta_description>
		<meta_keywords>
			<language id='1'>Meta</language>
		</meta_keywords>
		<associations>
			<tags>
				<tag>
					<id>106</id>
				</tag>
				<tag>
					<id>28</id>
				</tag>
				<tag>
					<id>297</id>
				</tag>
			</tags>
			<categories>
				<category>
					<id>24</id>
				</category>
			</categories>
		</associations>
	</product>
</prestashop>`

Response:
Product created successfully. 

Expected:
Product shouldn't be created. ﻿﻿

**Comment by WilfredChong** [comment]
Anyone? Please help

**Comment by khouloudbelguith** [comment]
Hi @WilfredChong,

Could you please try with the script to create the product.

```
<html><head><title>CRUD Tutorial - Create example</title></head><body>
<?php

define('DEBUG', true);
ini_set('display_errors','on');
define('PS_SHOP_PATH', 'http://shop/');
define('PS_WS_AUTH_KEY', 'Key');
require_once('PSWebServiceLibrary.php');

try {

    $webService= new PrestaShopWebservice(PS_SHOP_PATH, PS_WS_AUTH_KEY, DEBUG);
    $xml = $webService->get(array('url' => PS_SHOP_PATH.'/api/products?schema=blank'));

    $product = $xml->children()->children();
    $product->price = 99;
    $product->wholesale_price = 89;
    $product->active = '1';
    $product->advanced_stock_management = '1';
    $product->on_sale = 0;
    $product->show_price = 1;
    $product->available_for_order = 1;

    $product->name->language[0][0] = "Produit webservice test test 1750";
    $product->name->language[0][0]['id'] = 1;
    $product->name->language[0][0]['xlink:href'] = PS_SHOP_PATH . '/api/languages/' . 1;

    $product->description->language[0][0] = "Description produit webservice";
    $product->description->language[0][0]['id'] = 1;
    $product->description->language[0][0]['xlink:href'] = PS_SHOP_PATH . '/api/languages/' . 1;

    $product->description_short->language[0][0] = "Descr. courte";
    $product->description_short->language[0][0]['id'] = 1;
    $product->description_short->language[0][0]['xlink:href'] = PS_SHOP_PATH . '/api/languages/' . 1;
    $product->reference = "ref_product_webservice";
    $product->unit_price_ratio = 10.000000;
    $product->depends_on_stock = 0;

    $category_id = 2;
    $product->associations->categories->addChild('category')->addChild('id', $category_id);
    $product->id_category_default = $category_id;

    //
    $opt = array('resource' => 'products');
    $opt['postXml'] = $xml->asXML();
    $xml = $webService->add($opt);
}
catch (PrestaShopWebserviceException $ex) {
    echo "Error:<br>";
    echo  $ex->getMessage();
    exit(1);
}

?>
</body></html>
```
Thanks to check & feedback.

**Comment by WilfredChong** [comment]
Hi, where should i put this script?

**Comment by khouloudbelguith** [comment]
@WilfredChong, could you please followw this documentation:http://doc.prestashop.com/pages/viewpage.action?pageId=51184710
Thanks to check and feedback.

**Comment by WilfredChong** [comment]
Here
`<?xml version="1.0" encoding="UTF-8"?>
<prestashop xmlns:xlink="http://www.w3.org/1999/xlink">
<product>
	<id><![CDATA[1249]]></id>
	<id_manufacturer></id_manufacturer>
	<id_supplier></id_supplier>
	<id_category_default xlink:href="https://wako.com.my/api/categories/2"><![CDATA[2]]></id_category_default>
	<new></new>
	<cache_default_attribute></cache_default_attribute>
	<id_default_image notFilterable="true"></id_default_image>
	<id_default_combination notFilterable="true"></id_default_combination>
	<id_tax_rules_group></id_tax_rules_group>
	<position_in_category notFilterable="true"><![CDATA[1]]></position_in_category>
	<manufacturer_name notFilterable="true"></manufacturer_name>
	<quantity notFilterable="true"></quantity>
	<type notFilterable="true"><![CDATA[simple]]></type>
	<id_shop_default><![CDATA[1]]></id_shop_default>
	<reference><![CDATA[ref_product_webservice]]></reference>
	<supplier_reference></supplier_reference>
	<location></location>
	<width></width>
	<height></height>
	<depth></depth>
	<weight></weight>
	<quantity_discount></quantity_discount>
	<ean13></ean13>
	<upc></upc>
	<cache_is_pack></cache_is_pack>
	<cache_has_attachments></cache_has_attachments>
	<is_virtual></is_virtual>
	<on_sale><![CDATA[0]]></on_sale>
	<online_only></online_only>
	<ecotax></ecotax>
	<minimal_quantity></minimal_quantity>
	<price><![CDATA[99]]></price>
	<wholesale_price><![CDATA[89]]></wholesale_price>
	<unity></unity>
	<unit_price_ratio><![CDATA[10]]></unit_price_ratio>
	<additional_shipping_cost></additional_shipping_cost>
	<customizable></customizable>
	<text_fields></text_fields>
	<uploadable_files></uploadable_files>
	<active><![CDATA[1]]></active>
	<redirect_type></redirect_type>
	<id_product_redirected></id_product_redirected>
	<available_for_order><![CDATA[1]]></available_for_order>
	<available_date></available_date>
	<condition><![CDATA[new]]></condition>
	<show_price><![CDATA[1]]></show_price>
	<indexed></indexed>
	<visibility><![CDATA[both]]></visibility>
	<advanced_stock_management><![CDATA[1]]></advanced_stock_management>
	<date_add><![CDATA[2019-04-04 16:23:16]]></date_add>
	<date_upd><![CDATA[2019-04-04 16:23:16]]></date_upd>
	<pack_stock_type></pack_stock_type>
	<meta_description><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[]]></language></meta_description>
	<meta_keywords><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[]]></language></meta_keywords>
	<meta_title><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[]]></language></meta_title>
	<link_rewrite><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[produit-webservice-test-test-1750]]></language></link_rewrite>
	<name><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[Produit webservice test test 1750]]></language></name>
	<description><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[Description produit webservice]]></language></description>
	<description_short><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[Descr. courte]]></language></description_short>
	<available_now><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[]]></language></available_now>
	<available_later><language id="1" xlink:href="https://wako.com.my/api/languages/1"><![CDATA[]]></language></available_later>
<associations>
<categories nodeType="category" api="categories">
	<category xlink:href="https://wako.com.my/api/categories/2">
	<id><![CDATA[2]]></id>
	</category>
</categories>
<images nodeType="image" api="images"/>
<combinations nodeType="combination" api="combinations"/>
<product_option_values nodeType="product_option_value" api="product_option_values"/>
<product_features nodeType="product_feature" api="product_features">
	<product_feature xlink:href="https://wako.com.my/api/product_features/0">
	<id><![CDATA[0]]></id>
	<id_feature_value xlink:href="https://wako.com.my/api/product_feature_values/0"><![CDATA[0]]></id_feature_value>
	</product_feature>
</product_features>
<tags nodeType="tag" api="tags">
	<tag xlink:href="https://wako.com.my/api/tags/0">
	<id><![CDATA[0]]></id>
	</tag>
</tags>
<stock_availables nodeType="stock_available" api="stock_availables">
	<stock_available xlink:href="https://wako.com.my/api/stock_availables/2457">
	<id><![CDATA[2457]]></id>
	<id_product_attribute><![CDATA[0]]></id_product_attribute>
	</stock_available>
</stock_availables>
<accessories nodeType="product" api="products"/>
<product_bundle nodeType="product" api="products"/>
</associations>
</product>
</prestashop>`

**Comment by WilfredChong** [comment]
I created the product using that script, it still can be created the link rewrite with invalid field value like "abc.com", "vevev@#$rvev.jok"

**Comment by khouloudbelguith** [comment]
Hi @WilfredChong,

The format "abc.com" is not accepted in the "Friendly URL" field.
I attached a video record.
https://drive.google.com/file/d/1XztyUfAaokK3yz4GK1V8UZd_6A-ZfGgh/view
Thanks!
I close the ticket, feel free to open a new one when needed.

**Comment by WilfredChong** [comment]
I create it through the web server, it successful created when using POST method, but when using PUT method it returns link_rewrite not valid.

 And sorry I didn't mention that I am using Prestashop version 1.6

**Comment by khouloudbelguith** [comment]
@WilfredChong, as you can see here: the maintenance of PrestaShop 1.6 is extended to June 2019 for security and critical bugs.

**Comment by WilfredChong** [comment]
You mean the Prestashop is updating?

**Comment by WilfredChong** [comment]
Do I need to inform them about this issue?

**Comment by WilfredChong** [comment]
Hello? anyone?

**Comment by khouloudbelguith** [comment]
Hi @WilfredChong,

The maintenance of PrestaShop 1.6 except for security and critical bugs.

Thanks!

**Comment by matks** [comment]
> You mean the Prestashop is updating?

You are using an old version of prestashop (PS 1.6), which is under extended maintenance. This means that only critical bugs and security bugs are fixed for PS 1.6 (in order to keep it as stable as possible) . Your issue seems fixed in the latest PrestaShop version (1.7).
